packages/redux/src/sizePreferences/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,165 @@
+import * as actionTypes from '../actionTypes.js';
+import { type ListingFacet, toBlackoutError } from '@farfetch/blackout-client';
+import {
+  mockListingFacets,
+  mockQuery,
+} from 'tests/__fixtures__/sizePreferences/index.mjs';
+import reducer, * as fromReducer from '../reducer.js';
+import type { SizePreferencesState } from '../types/state.types.js';
+
+let initialState: SizePreferencesState;
+const randomAction = { type: 'this_is_a_random_action' };
+
+describe('sizePreferences reducer', () => {
+  beforeEach(() => {
+    initialState = fromReducer.INITIAL_STATE;
+  });
+
+  describe('reset handling', () => {
+    it('should return the initial state', () => {
+      expect(
+        reducer(undefined, {
+          type: actionTypes.RESET_SIZE_PREFERENCES_STATE,
+        }),
+      ).toEqual(initialState);
+    });
+  });
+
+  describe('error() reducer', () => {
+    it('should return the initial state', () => {
+      const state = reducer(undefined, randomAction).error;
+
+      expect(state).toEqual(initialState.error);
+      expect(state).toBeNull();
+    });
+
+    it('should handle FETCH_LISTING_FACETS_REQUEST action type', () => {
+      const expectedResult = initialState.error;
+
+      expect(
+        reducer(undefined, {
```

Comment:
Please, do not use `undefined` in these tests as this is already covered by the `should return the initial state` test. Use a state that is representative for when this action happens (like isLoading: false and error set to a value).